### DISCUSSION TITLE
predicting more than one set of features

### DISCUSSION THREAD

**Discussion by duduhqp** [online-ml/river#828] [Q&A] [answered]
Hi, i'm very new to this area and i'm trying to implement a SAM-kNN classifier, but the only implemented method for predicition is predict_one? So I have to implement a predict_more_than_one sort of thing ? How would that go?

If it's already implemented it doesn't say in the API reference.

Thanks for the help.

**Comment by raphaelsty** [comment, marked as answer]
Hi @duduhqp 😀,

River's philosophy is to offer highly optimised algorithms for learning and predicting one observation at a time. Some River models are able to train and predict on several observations at once. These models have a dedicated and optimized `learn_many` and `predict_many` method.  Unfortunately SAM-KNN does not have such method yet. You could rely on a for loop to predict over several samples

**Reply by raphaelsty** [reply]
@duduhqp I'll close this discussion, feel free to ask further questions :).
